lib/src/json_schema/constants.dart

```diff
@@ -46,13 +55,24 @@ class JsonSchemaValidationRegexes {
       r'(\d|[1-9]\d|1\d\d|2([0-4]\d|5[0-5]))\.'
       r'(\d|[1-9]\d|1\d\d|2([0-4]\d|5[0-5]))$');
 
+  // Spec: https://datatracker.ietf.org/doc/html/rfc1034
   static RegExp hostname = RegExp(r'^(?=.{1,255}$)'
       r'[0-9A-Za-z](?:(?:[0-9A-Za-z]|-){0,61}[0-9A-Za-z])?'
       r'(?:\.[0-9A-Za-z](?:(?:[0-9A-Za-z]|-){0,61}[0-9A-Za-z])?)*\.?$');
 
+  // Spec: https://datatracker.ietf.org/doc/html/rfc1123
+  static RegExp hostnameDraft2019 = RegExp(r'^(?=.{1,255}$)'
+      r'[0-9A-Za-z](?:(?:[0-9A-Za-z]|-){0,61}[0-9A-Za-z])?'
+      r'(?:\.[0-9A-Za-z](?:(?:[0-9A-Za-z]|-){0,61}[0-9A-Za-z])?)*\.?$');
```

Comment:
I split these apart since technically they _ought to be_ different. However, as it turns out, our existing regex seems to allow the leading digit that was newly allowed in 2019. So probably our Draft 7 regex is incorrect :man_shrugging: